record/custom_records.go

```diff
@@ -22,3 +27,62 @@ func (c CustomSet) Validate() error {
 
 	return nil
 }
+
+// Encode writes the CustomSet to the provided io.Writer in big endian format.
+func (c *CustomSet) Encode(w io.Writer) error {
+	for key, value := range *c {
```

Comment:
Should we also write the number of entries in the map so we don't have to rely on EOF when reading?

Reply:
Removed in latest commits.